arches/app/src/components/ControlledListManager/ListTreeControls.vue

```diff
@@ -0,0 +1,425 @@
+    if (!itemIds.length) {
+        return;
+    }
+    const token = Cookies.get("csrftoken");
+    if (!token) {
+        return;
+    }
+    const promises = itemIds.map((id) =>
+        fetch(arches.urls.controlled_list_item(id), {
+            method: "DELETE",
+            headers: { "X-CSRFToken": token },
+        })
+    );
+
+    try {
+        const responses = await Promise.all(promises);
+        if (responses.some((resp) => resp.ok)) {
+            setDisplayedRow(null);
+        }
+        responses.forEach(async (response) => {
+            if (!response.ok) {
+                const body = await response.json();
+                toast.add({
+                    severity: ERROR,
+                    life: 8000,
+                    summary: $gettext("Item deletion failed"),
+                    detail: body.message,
+                });
+            }
+        });
+    } catch {
+        toast.add({
+            severity: ERROR,
+            life: 8000,
+            summary: $gettext("Item deletion failed"),
+        });
+    }
+};
+
+const toDelete = computed(() => {
+    if (isMultiSelecting.value) {
+        return Object.entries(selectedKeys.value).filter(([, v]) => v.checked).map(([k,]) => k);
+    }
+    return Object.entries(selectedKeys.value).filter(([, v]) => v).map(([k,]) => k);
+});
+
+const deleteSelected = async () => {
+    if (!selectedKeys.value) {
+        return;
+    }
+    const allListIds = controlledListItemsTree.value.map((node: TreeNode) => node.data.id);
+
+    const listIdsToDelete = toDelete.value.filter(id => allListIds.includes(id));
+    const itemIdsToDelete = toDelete.value.filter(id => !listIdsToDelete.includes(id));
+
+    selectedKeys.value = {};
+
+    // Do items first so that cascade deletion doesn't cause item deletion to fail.
+    await deleteItems(itemIdsToDelete);
+    await deleteLists(listIdsToDelete);
+
+    isMultiSelecting.value = false;
+};
+
+const confirmDelete = () => {
+    const numItems = toDelete.value.length;
+    confirm.require({
+        message: $ngettext(
+            "Are you sure you want to delete %{numItems} item (including all children)?",
+            "Are you sure you want to delete %{numItems} items (including all children)?",
+            numItems,
+            { numItems: numItems.toLocaleString() },
+        ),
+        header: $gettext("Confirm deletion"),
+        icon: "fa fa-exclamation-triangle",
+        rejectLabel: $gettext("Cancel"),
+        rejectClass: "p-button-secondary p-button-outlined",
+        acceptLabel: $gettext("Delete"),
+        accept: async () => {
+            await deleteSelected().then(fetchLists);
+        },
+        reject: () => {},
+    });
+};
+
+const abandonMove = () => {
+    movingItem.value = {};
+
+    // Clear custom classes added in <ListTree> pass-through
+    Array.from(
+        abandonMoveRef.value!.$el.ownerDocument.getElementsByClassName('is-adjusting-parent')
```

Comment:
thanks, I think I shied away from that worrying it would issue more network requests, but that's not the case and forcing a re-render does the trick 🎉